program/src/processor.rs

```diff
@@ -560,9 +558,15 @@ fn process_harvest_rewards(program_id: &Pubkey, accounts: &[AccountInfo]) -> Pro
             .unharvested_rewards
             .checked_sub(rewards_to_harvest)
             .ok_or(ProgramError::ArithmeticOverflow)?;
-    }
 
-    Ok(())
+        // Update the holder's last seen rate.
+        holder_rewards_state.last_accumulated_rewards_per_token =
+            pool_state.accumulated_rewards_per_token;
+
+        Ok(())
+    } else {
+        Err(PaladinRewardsError::NoRewardsToHarvest.into())
+    }
```

Comment:
Since `HarvestRewards` is permissionless, it might make sense to assume that someone will crank it for others, in which case they might bundle a lot of instructions in one transaction. If they do that, it would stink for a transaction to fail because one of them doesn't actually have rewards to harvest.

I lean towards not having the error.

Reply:
I feel like we talked about this and landed on _not_ doing the error, but I must have a brain injury.